### PR TITLE
Add per-site Xrootd statistics to the HTCondor monitoring.

### PR DESCRIPTION
**Pull request by bbockelm**
Per popular demand, this will allow us to see per-site bytes read and read time, aggregated by site (as identified by the remote server; may be different from the CMS site name).

@osschar

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: Utilities/XrdAdaptor/src/XrdStatistics.h [hunk 1]
```diff
@@ -41,8 +46,16 @@ class XrdStatisticsService
...
+    struct CondorIOStats {
+      uint64_t bytesRead{0};
+      std::chrono::nanoseconds transferTime{0};
+    };
 
+    // Provide an update of per-site transfer statistics to the CondorStatusService.
+    // Returns a mapping of "site name" to transfer statistics.  The "site name" is
+    // as self-identified by the Xrootd host; may not necessarily match up with the
+    // "CMS site name".
+    std::map<std::string, CondorIOStats> condorUpdate();
```
Comment (Dr15Jones) [inline Utilities/XrdAdaptor/src/XrdStatistics.h]: A `std::vector<std::pair<std::string, CondorIOStats>` would be more memory efficient and probably faster to fill.

File: Utilities/XrdAdaptor/src/XrdStatistics.cc [hunk 2]
```diff
@@ -44,6 +44,26 @@ void XrdStatisticsService::postEndJob()
...
 }
 
+std::map<std::string, XrdStatisticsService::CondorIOStats>
+XrdStatisticsService::condorUpdate()
+{
+    std::map<std::string, XrdStatisticsService::CondorIOStats> result;
+    XrdSiteStatisticsInformation *instance = XrdSiteStatisticsInformation::getInstance();
+    if (!instance) {return result;}
+
+    std::lock_guard<std::mutex> lock(instance->m_mutex);
```
Comment (Dr15Jones) [inline Utilities/XrdAdaptor/src/XrdStatistics.cc]: For future: exposing a lock is a poor interface. We should rethink this service.

File: FWCore/Framework/src/defaultCmsRunServices.cc [hunk 3]
```diff
@@ -23,7 +23,8 @@ namespace edm {
                                               "AdaptorConfig",
                                               "SiteLocalConfigService",
                                               "StatisticsSenderService",
-                                              "CondorStatusService"};
+                                              "CondorStatusService",
+                                              "XrdAdaptor::XrdStatisticsService"};
```
Comment (Dr15Jones) [inline FWCore/Framework/src/defaultCmsRunServices.cc]: I'm not overly happy about adding more default services to jobs where the framework itself does not need those services to operate.

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
@bbockelm waiting on a reply to my comments

**Comment by bbockelm** [comment]
@Dr15Jones - I tackled the item that had a concrete action item; however, it's not clear from your other comments what you'd like me to do (or if they're just notes for the future).

**Comment by davidlange6** [comment]
please test

**Comment by Dr15Jones** [comment]
+1
We need to keep an eye on this since we are now causing all test configurations of cmsRun to force loading of xrootd client.

File: FWCore/Framework/src/defaultCmsRunServices.cc (hunk 3 above)
Comment (davidlange6) [reply FWCore/Framework/src/defaultCmsRunServices.cc]: hi @dr15jones - perhaps you need two lists here - one for the things the FWK needs and one for the things that CMS wants by default?

**Comment by davidlange6** [comment]
+1

**Pull request merged**